src/db/models/location.ts

```diff
@@ -10,3 +11,39 @@ export type LocationId = Brand<
 >;
 
 export const LOCATION_ID = brandedType<number, LocationId>(t.number);
+
+const LOCATION_STATUS = {
+  active: null,
+  expired: null,
+};
+
+export default defineIDModel({
+  tableName: 'location',
+  fields: {
+    generated: {
+      id: { kind: 'branded-integer', brand: LOCATION_ID },
+    },
+    optional: {
+      externalId: { kind: 'checked', type: t.string },
+      name: { kind: 'checked', type: t.string },
+      latitude: { kind: 'checked', type: t.number },
+      longitude: { kind: 'checked', type: t.number },
+      iso3: { kind: 'checked', type: t.string },
+      pcode: { kind: 'checked', type: t.string },
+      // Even though this column is defined as int8, it is
+      // fetched as a string by knex, since it is bigint
+      validOn: { kind: 'checked', type: t.string },
+      parentId: { kind: 'branded-integer', brand: LOCATION_ID },
+    },
+    accidentallyOptional: {
+      adminLevel: { kind: 'checked', type: t.number },
+      status: {
+        kind: 'enum',
+        values: LOCATION_STATUS,
+      },
```

Comment:
`enum` should only be used when the property uses an enum defined in sql, this particular field is a string, so instead we should use a checked string with `t.keyof` to validate upon read, and restrict the type when writing.

The reason for this is that at some point I'd like to extend the validation function / unit tests to check that enum types are defined correctly, and allow the same values.

Reply:
Makes sense even now, without enum definition checks